tests/event-bus/e2e-tester/e2e-tester.go

```diff
@@ -333,90 +321,114 @@ func publishTestEvent(publishEventURL string) (*api.Response, error) {
+	if err != nil {
+		return nil, err
+	}
 	err = json.Unmarshal(body, &respObj)
 	if err != nil {
-		log.Printf("Unmarshal error: %v", err)
 		return nil, err
 	}
-	log.Printf("Publish response object: %+v", respObj)
+	log.WithField("response", respObj).Info("publish response object")
 	if len(respObj.EventID) == 0 {
 		return nil, fmt.Errorf("empty respObj.EventID")
 	}
 	return respObj, err
 }
 
 func (subDetails *subscriberDetails) checkSubscriberReceivedEvent() error {
-	for i := 0; i < retries; i++ {
-		time.Sleep(time.Duration(i) * time.Second)
-		log.Printf("Get subscriber response (%d/%d)\n", i, retries)
+	return retry.Do(func() error {
 		res, err := http.Get(subDetails.subscriberResultsEndpointURL)
 		if err != nil {
-			log.Printf("Get request failed: %v\n", err)
 			return err
 		}
-		dumpResponse(res)
+		defer func() {
+			if err := res.Body.Close(); err != nil {
+				log.Warn(err)
+			}
+		}()
+		if _, err := httputil.DumpResponse(res, true); err != nil {
+			return err
+		}
 		if err := verifyStatusCode(res, 200); err != nil {
-			log.Printf("Get request failed: %v", err)
 			return err
 		}
 		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
 		var resp string
-		json.Unmarshal(body, &resp)
-		log.Printf("Subscriber response: %s\n", resp)
-		res.Body.Close()
-		if len(resp) == 0 { // no event received by subscriber
-			continue
+		err = json.Unmarshal(body, &resp)
+		if err != nil {
+			return err
+		}
+		log.WithField("response", resp).Info("subscriber response")
+		if len(resp) == 0 {
+			return errors.New("no event received by subscriber")
 		}
 		if resp != "test-event-1" {
 			return fmt.Errorf("wrong response: %s, want: %s", resp, "test-event-1")
 		}
 		return nil
-	}
-	return errors.New("timeout for subscriber response")
+	}, retryOptions...)
 }
 
 func (subDetails *subscriberDetails) checkSubscriberReceivedEventHeaders() error {
-	for i := 0; i < retries; i++ {
-		time.Sleep(time.Duration(i) * time.Second)
-		log.Printf("Get subscriber 3 response (%d/%d)\n", i, retries)
+	return retry.Do(func() error {
 		res, err := http.Get(subDetails.subscriber3ResultsEndpointURL)
 		if err != nil {
-			log.Printf("Get request failed: %v\n", err)
 			return err
 		}
-		dumpResponse(res)
+		defer func() {
+			if err := res.Body.Close(); err != nil {
+				log.Warn(err)
+			}
+		}()
+		if _, err := httputil.DumpResponse(res, true); err != nil {
+			return err
+		}
 		if err := verifyStatusCode(res, 200); err != nil {
-			log.Printf("Get request failed: %v", err)
 			return err
 		}
 		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
 		var resp map[string][]string
-		json.Unmarshal(body, &resp)
-		log.Printf("Subscriber 3 response: %v\n", resp)
-		res.Body.Close()
-		if len(resp) == 0 { // no event received by subscriber
-			continue
+		if err := json.Unmarshal(body, &resp); err != nil {
+			return err
+		}
+		log.WithField("response", resp).Info("subscriber 3 response")
```

Comment:
Can we have something like this as `response for v3 endpoint of the subscriber"`?